lib/shopify_api/conn_helpers.ex

```diff
@@ -10,32 +10,46 @@ defmodule ShopifyAPI.ConnHelpers do
   @shopify_hmac_header "x-shopify-hmac-sha256"
 
   @doc false
-  def fetch_shopify_app(conn), do: fetch_shopify_app(conn, app_name(conn))
-  def fetch_shopify_app(_conn, app_name), do: AppServer.get(app_name)
+  defp optionally_create_shop(:error, conn), do: {:ok, %Shop{domain: shop_domain(conn)}}
+  defp optionally_create_shop(shop, _), do: shop
 
   @doc false
-  def fetch_shopify_shop(conn), do: fetch_shopify_shop(conn, shop_domain(conn))
-
-  def fetch_shopify_shop(conn, shop_domain),
-    do: shop_domain |> ShopServer.get() |> optionally_create_shop(conn)
+  def hmac_from_header(conn) do
+    conn
+    |> Conn.get_req_header(@shopify_hmac_header)
+    |> List.first()
+  end
 
   @doc false
-  defp optionally_create_shop(:error, conn), do: {:ok, %Shop{domain: shop_domain(conn)}}
-  defp optionally_create_shop(shop, _), do: shop
+  def auth_code(conn), do: conn.params["code"]
 
   @doc false
   def app_name(conn), do: conn.params["app"] || app_name_from_path(conn)
 
+  @doc false
   def app_name_from_path(conn), do: List.last(conn.path_info)
 
   @doc false
-  def shop_domain(conn), do: shop_domain_from_header(conn) || conn.params["shop"]
+  def fetch_shopify_app(conn), do: conn |> app_name() |> AppServer.get()
 
   @doc false
-  def hmac_from_header(conn) do
-    conn
-    |> Conn.get_req_header(@shopify_hmac_header)
-    |> List.first()
+  def assign_app(conn, app_name \\ nil)
+  def assign_app(conn, %App{} = app), do: Conn.assign(conn, :app, app)
+
+  def assign_app(conn, app_name) do
+    app_name =
+      case app_name do
+        nil -> app_name(conn)
```

Comment:
This is confusing to read - could we disambiguate `app_name` (the parameter) from `app_name` (the function)?

Reply:
yea, good call.  That was bugging me earlier and didn't do anything